File: src/main/java/com/naminhyeok/fantazzk/template/TemplateExceptionHandler.java
package com.naminhyeok.fantazzk.template;

import com.naminhyeok.fantazzk.ApiResponse;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice(basePackageClasses = TemplateApiController.class)
public class TemplateExceptionHandler {
    @ExceptionHandler(TemplateNotFoundException.class)
    ResponseEntity<ApiResponse<Void>> handleNotFound(TemplateNotFoundException ex) {
        return ResponseEntity.status(HttpStatus.NOT_FOUND)
            .body(ApiResponse.error(HttpStatus.NOT_FOUND.value(), "TEMPLATE_NOT_FOUND", ex.getMessage()));
    }

    @ExceptionHandler(IllegalArgumentException.class)
    ResponseEntity<ApiResponse<Void>> handleBadRequest(IllegalArgumentException ex) {
        return ResponseEntity.status(HttpStatus.BAD_REQUEST)
            .body(ApiResponse.error(HttpStatus.BAD_REQUEST.value(), "BAD_REQUEST", ex.getMessage()));
    }
}

File: template/api/src/main/kotlin/com/naminhyeok/fantazzk/template/TemplateExceptionHandler.kt
package com.naminhyeok.fantazzk.template

import com.naminhyeok.fantazzk.template.exception.TemplateNotFoundException
import org.springframework.http.HttpStatus
import org.springframework.http.ProblemDetail
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice(basePackageClasses = [TemplateExceptionHandler::class])
class TemplateExceptionHandler {
    @ExceptionHandler(TemplateNotFoundException::class)
    fun handleNotFound(ex: TemplateNotFoundException): ProblemDetail =
        ProblemDetail.forStatusAndDetail(HttpStatus.NOT_FOUND, ex.message ?: "Not found")

    @ExceptionHandler(IllegalArgumentException::class)
    fun handleBadRequest(ex: IllegalArgumentException): ProblemDetail =
        ProblemDetail.forStatusAndDetail(HttpStatus.BAD_REQUEST, ex.message ?: "Bad request")
}
